query/logicalplan/builder.go

```diff
@@ -237,6 +237,22 @@ func resolveAggregation(plan *LogicalPlan, agg *AggregationFunction) ([]*Aggrega
 	}
 }
 
+func (b Builder) Sample(expr Expr) Builder {
```

Comment:
We should make the type of sampler used configurable, while we currently just need the reservoir sampler, there can certainly be valid use cases for others.

Reply:
Yea I waffled on that since we didn't have a use case for it today I decided to punt on it and not add complexity yet.